### PR TITLE
Example Enable/Disable functionality for PIDController

### LINKED ISSUE #833: Document enable/disable functionality for PIDController widget
The PIDController widget for SmartDashboard/Shuffleboard has an enabled boolean property that isn't supported by the new PIDController class. While it was very important to be able to disable the old PIDController since it automatically wrote to the motor, it is still helpful to pass the enabled flag to user code for while tuning with LiveWindow.

### PR DESCRIPTION
**Pull request by jasondaming**
Closes #833

### CODE REVIEW

File: source/docs/software/wpilib-tools/shuffleboard/advanced-usage/shuffleboard-tuning-pid.rst [hunk 1]
```diff
@@ -28,6 +28,34 @@ This is the test mode picture of a wrist subsystem that has a potentiometer as t
 4. The setpoint value that corresponds the to the pot value when the wrist has reached the desired value
 5. Enables the PID controller - that is, starts it looping at regular intervals reading the pot value, computing the error, applying the P, I, and D terms and setting the motor value.
 
-.. important:: The enable option does not affect the `PIDController <https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/controller/PIDController.html>`__ introduced in 2020, as the controller is updated every robot loop. Users who wish to retain this functionality can retrieve the value from NetworkTables and implement the feature themselves.
+.. important:: The enable option does not affect the `PIDController <https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/controller/PIDController.html>`__ introduced in 2020, as the controller is updated every robot loop. See the example below on how to retain this fuctionality.
+
+Implementing Enable Functionality in the New PIDController
```
Comment (Daltz333) [inline source/docs/software/wpilib-tools/shuffleboard/advanced-usage/shuffleboard-tuning-pid.rst:33]: This title is particularly long, and doesn't wrap well on mobile. Can we shorten this?

Comment (jasondaming) [reply source/docs/software/wpilib-tools/shuffleboard/advanced-usage/shuffleboard-tuning-pid.rst:33]: what about "Enable Functionality in the New PIDController" or is that still too long?

**Review by prateekma — CHANGES_REQUESTED** [review]
There is no actual `enable()` method in `PIDController` for `m_controller::enable` to work. The `NetworkButton` should set some other boolean value and the user should choose to call `m_controller.calculate()` based on the state of that value.

**Comment by jasondaming** [comment]
Ahh that makes more sense.  In the places where I am using PIDController I am using a PIDSubsystem which does have enable (I guess because it is running or not running the calculate under the hood like this)

**Comment by jasondaming** [comment]
I am not sure how I fix this.  Can this be assumed / understood:
`if (shooterEnable.getBoolean()) {
      m_controller.calculate();
 }`
If so is the C++ pretty much the same?

**Comment by Starlight220** [comment]
`controller.calculate()` has a `measurement` parameter that you're forgetting.  C++ methods are in UpperCamelCase (`controller.Calculate()` etc). Other than that, yes.

**Comment by jasondaming** [comment]
hopefully everything is correct now but someone should probably double check my C++ to make sure

File: source/docs/software/wpilib-tools/shuffleboard/advanced-usage/shuffleboard-tuning-pid.rst [hunk 2]
```diff
@@ -26,8 +26,45 @@ This is the test mode picture of a wrist subsystem that has a potentiometer as t
...
+  .. code-tab:: c++
+
+    frc::ShuffleboardTab& tab = frc::Shuffleboard::GetTab("Shooter");
+    nt::NetworkTableEntry shooterEnable = tab.Add("Shooter Enable", false).GetEntry();
+
+    // Command-based assumed to be in a PIDSubsystem
+    frc2::NetworkButton(shooterEnable).WhenPressed(frc2::InstantCommand([&] { m_shooter.Enable(); }));
+
+    // Timed Robot Example
+    if (shooterEnable.getBoolean()) {
```
Comment (prateekma) [inline source/docs/software/wpilib-tools/shuffleboard/advanced-usage/shuffleboard-tuning-pid.rst:66]: Should be `GetBoolean()`

**Review by prateekma — CHANGES_REQUESTED** [review]

File: source/docs/software/wpilib-tools/shuffleboard/advanced-usage/shuffleboard-tuning-pid.rst (hunk 2 above)
Comment (jasondaming) [reply source/docs/software/wpilib-tools/shuffleboard/advanced-usage/shuffleboard-tuning-pid.rst]: thanks fixed

**Review by prateekma — APPROVED** [review]

**Review by Daltz333 — APPROVED** [review]

**Pull request merged**